controllers/tasks.js

```diff
@@ -16,18 +17,36 @@ const { INTERNAL_SERVER_ERROR, SOMETHING_WENT_WRONG } = require("../constants/er
 const addNewTask = async (req, res) => {
   try {
     const { id: createdBy } = req.userData;
+    const dependsOn = req.body.dependsOn;
     const body = {
       ...req.body,
       createdBy,
     };
-
-    const task = await tasks.updateTask(body);
-
-    return res.json({
-      message: "Task created successfully!",
-      task: task.taskDetails,
-      id: task.taskId,
-    });
+    delete body.dependsOn;
+    const { taskId, taskDetails } = await tasks.updateTask(body);
+    const data = {
+      taskId,
+      dependsOn,
+    };
+    if (data.dependsOn) {
+      const taskDependency = await dependencyModel.addDependency(data);
+      return res.json({
+        message: "Task created successfully!",
+        task: {
+          ...taskDetails,
+          dependsOn: taskDependency,
+          id: taskId,
+        },
+      });
+    } else {
+      return res.json({
+        message: "Task created successfully!",
+        task: {
+          ...taskDetails,
+          id: taskId,
+        },
+      });
+    }
```

Comment:
I try but both task creation site site is different approaches and I see the tag controller to add a tag has conditional res.json

Reply:
![image](https://github.com/Real-Dev-Squad/website-backend/assets/111434418/86619966-cd9e-45cc-9fba-bffe5a01f871)
this can be a solution for more than 500 document

Reply:
try to check in your local by setting the limit say 2 and try to add 3 docs. if it works then this 500 doc check will be good.

Reply:
> I try but both task creation site is different approaches and I see the tag controller to add a tag has conditional res.json this code is work well should I apply this ![image](https://user-images.githubusercontent.com/111434418/237880715-83e844b1-898f-465a-9d5c-130dfc990f17.png)

this still has 2 return keywords. try to handle both in a single return statement

Reply:
> try to check in your local by setting the limit say 2 and try to add 3 depends on tasks. if it works then this 500 doc check will be good.

I done it and it will work

Reply:
```js
task : {
    ...taskDetails,
    ...(taskDependancy && {dependsOn : taskDependancy}),
    id: taskId
}